### ISSUE TITLE
set up electron.js

### ISSUE THREAD

**Issue by nickehsani** [cse110-fa22-group13/cse110-fa22-group13#40]
- Set up electron.js
- Give feedback on it
- Create documentation (.readme on how to launch the app)
Goal: have electronjs working

**Comment by Donttestturing** [comment]
We have a sample app ready to go, it is a simpler version of the expose portion of lab5. (No confetti) It was relatively straightforward to package a HTML, JS, CSS webpage app into ElectronJS. However, I foresee that we might have some trouble with the local storage stuff, as that seems to be an additional layer of complexity, but I am unsure about that at this stage. There is the idea that we mostly interact with the renderer, and Node.js is behind the scenes, and there isn't usually direct access between the two, for security reasons. This is the section of a tutorial I am talking about. https://www.electronjs.org/docs/latest/tutorial/tutorial-preload.

We will have a main.js file, which encapsulates the app and does the packaging. And then we will have separate script file(s), html file(s), css file(s) as well. 

I will note even this small app, the files of which were about 700 kb, the exe is now 153000 kb. It grew gigantically.

**Comment by Donttestturing** [comment]
We are now exploring alternatives to ElectronJS, Cordova and Tauri

**Comment by Donttestturing** [comment]
One of the repos is here: https://github.com/Donttestturing/testElectronApp

**Comment by adaburton** [comment]
During my research I found that in a perfect world, we might use Tauri. It solves many of the drawbacks that come from wrapping an app with electron, by replacing alot of the functionality previously handled by a chromium browser engine with rust. This would allow our app to be much smaller, faster, and have lower level control with the operating system, and we would get the added benefit of our app feeling more native to whichever OS it is running on. However I think this technology is beyond the scope of the class, especially with what little time we have left in the quarter. It also forces us to use rust, which I am familiar with, but also lies outside of the scope of the class. 

Cordova I found to be more or less equivalent to Electron, with slightly less intuitive documentation. 

I think electronJS is still the best option for our specific situation.
